diesel/src/macros/mod.rs

```diff
@@ -228,108 +252,248 @@ macro_rules! __diesel_column {
 /// ```
 #[macro_export]
 macro_rules! table {
-    // Put `use` statements at the end because macro_rules! cannot figure out
-    // if `use` is an ident or not (hint: It's not)
+    // Put imports into the import field
     (
-        use $($import:tt)::+; $($rest:tt)+
+        @parse
+        import = [$(use $($import:tt)::+;)*];
+        doc = [$($doc:expr,)*];
```

Comment:
In this case you are right, we do not need to parse doc here (It's removed in my last changes).

In general:
We need to match those `*_doc =[$($doc:expr)*]` things in a repeater because:
 - There may be none
 - A comment may contain multiple lines:
```rust 
/// First line
/// Second line
 ```
   Rustc desugers this into:
```rust
#[doc=r"First line"]
#[doc=r"Second line"]
```
Therefore we must match on multiple doc attributes